src/controllers/recommendationController.js

```diff
@@ -0,0 +1,23 @@
+const AlbumRecommendation = require('../models/AlbumRecommendation');
+const recommendationController = require('../controllers/recommendationController');
+async function generateAlbumRecommendations(userId) {
+  return await AlbumRecommendation.find({ userId });
+}
+const getAlbumRecsForUser = async (req, res) => {
+  const { userId } = req.params;
+  try {
+    const recommendations =
+      await recommendationController.generateAlbumRecommendations(userId);
```

Comment:
At this point, having `generateAlbumRecommendations` as a separate function doesn't add too much and may add a bit of misdirection. Let's refactor so that the call to the model is inline here again instead of a calling a function to call a function 😄

Reply:
Refactoring done 😊